src/Symfony/Bridge/Doctrine/ConnectionTimingWeakMap.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine;
+
+class ConnectionTimingWeakMap
+{
+    private ?\WeakMap $connectionWeakMap = null;
+
+    public function __construct()
+    {
+        $this->connectionWeakMap = $this->connectionWeakMap ?? new \WeakMap();
```

Comment:
`$this->connectionWeakMap` should not be nullable, because you put it right here. And checking beforehand whether the property is set is not necessary in a constructor. You can also use phpdocs to specify the data type of the WeapMap: `WeakMap<Connection, array{timing: int, name: string}>`